ellie/monopoly.py

```diff
@@ -0,0 +1,154 @@
+        self.rent5 = property_rent5
+        self.mortgage = property_mortgage
+    
+    def __repr__(self):
+        return f"{self.name}: ${self.cost}. \nRent ${self.rent0}. \nWith 1 House: ${self.rent1}. \nWith 2 Houses: ${self.rent2}. \nWith 3 Houses: ${self.rent3}. \nWith 4 Houses: ${self.rent4}. \nWith Hotel: ${self.rent5}. \n\nMortgage Value: ${self.mortgage}. \nHouses cost ${self.house_cost} each. \nHotels, ${self.house_cost} plus 4 houses. \nIf a player owns ALL the lots in any Color-Group, the rent is Doubled on Unimproved Lots in that group."
+    
+    def buy_property(self, player):
+        if player.money < self.cost:
+            print(f"You don't have enough money to afford this property.")
+            return
+        else:
+            player.properties.append(self)
+
+    def buy_house(self, player):
+        if player.money < self.house_cost:
+            print(f"You don't have enough money to afford a house here.")
+            return
+        else:
+            if self.houses < 5:
+                self.houses += 1
+            else:
+                print(f"This property already has a hotel.")
+        
+    def charge_rent(self, owner, renter):
+        if self.houses <= 0:
+            owner.money += self.rent0
+            renter.money -= self.rent0
+            print(f"You landed on {self.name}. Rent with {self.houses} houses costs ${self.rent0}.")
+            print(f"{renter.name} paid {owner.name} ${self.rent0}.")
+        elif self.houses == 1:
+            owner.money += self.rent1
+            renter.money -= self.rent1
+            print(f"You landed on {self.name}. Rent with {self.houses} houses costs ${self.rent1}.")
+            print(f"{renter.name} paid {owner.name} ${self.rent1}.")
+        elif self.houses == 2:
+            owner.money += self.rent2
+            renter.money -= self.rent2
+            print(f"You landed on {self.name}. Rent with {self.houses} houses costs ${self.rent2}.")
+            print(f"{renter.name} paid {owner.name} ${self.rent2}.")
+        elif self.houses == 3:
+            owner.money += self.rent3
+            renter.money -= self.rent3
+            print(f"You landed on {self.name}. Rent with {self.houses} houses costs ${self.rent3}.")
+            print(f"{renter.name} paid {owner.name} ${self.rent3}.")
+        elif self.houses == 4:
+            owner.money += self.rent4
+            renter.money -= self.rent4
+            print(f"You landed on {self.name}. Rent with {self.houses} houses costs ${self.rent4}.")
+            print(f"{renter.name} paid {owner.name} ${self.rent4}.")
+        elif self.houses >= 5:
+            owner.money += self.rent5
+            renter.money -= self.rent5
+            print(f"You landed on {self.name}. Rent with 1 hotel costs ${self.rent5}.")
+            print(f"{renter.name} paid {owner.name} ${self.rent5}.")
+
+mediterranean_ave = Property("Mediterranean Avenue", 60, 50, 2, 10, 30, 90, 160, 250, 30)
```

Comment:
When a function has lots of arguments it can be difficult at a glance to figure out which is which without memorizing and re-memorizing the order of arguments. Python lets you use keyword arguments even when you don't _have_ to, which can often make things more readable at the cost of slightly more typing:

```suggestion
mediterranean_ave = Property(
    property_name="Mediterranean Avenue", 
    property_cost=60, 
    property_house_cost=50, 
    property_rent0=2, 
    property_rent1=10, 
    property_rent2=30, 
    property_rent3=90, 
    property_rent4=160, 
    property_rent5=250, 
    property_mortgage=30,
)
```